zetta_utils/mazepa_layer_processing/common/volumetric_apply_flow.py

```diff
@@ -577,11 +587,22 @@ def flow(  # pylint:disable=too-many-branches, too-many-statements
                 copy_chunk_size = dst.backend.get_chunk_size(self.dst_resolution)
             else:
                 copy_chunk_size = self.max_reduction_chunk_size_final
-            reduction_chunker = VolumetricIndexChunker(
-                chunk_size=dst.backend.get_chunk_size(self.dst_resolution),
-                resolution=self.dst_resolution,
-                max_superchunk_size=copy_chunk_size,
-            )
+
+            if self.gap is None or self.gap == Vec3D[int](0, 0, 0):
+                reduction_chunker = VolumetricIndexChunker(
+                    chunk_size=dst.backend.get_chunk_size(self.dst_resolution),
+                    resolution=self.dst_resolution,
+                    max_superchunk_size=copy_chunk_size,
+                )
+            else:
+                stride_and_chunk_size = (
+                    dst.backend.get_chunk_size(self.dst_resolution) - self.gap // 2
+                )
+                reduction_chunker = VolumetricIndexChunker(
+                    chunk_size=stride_and_chunk_size,
+                    resolution=self.dst_resolution,
+                    max_superchunk_size=stride_and_chunk_size,
+                )
```

Comment:
Same as above - I think we can get rid of this if-else block. Also: 
- `max_superchunk_size` should always be `copy_chunk_size`.
- `offset` should be set to `- self.gap // 2`.